src/index.ts

```diff
@@ -275,7 +279,12 @@ export function register (options: Options = {}): Register {
             return undefined
           }
 
-          cache.contents[fileName] = getFile(fileName)
+          const content = getFile(fileName)
+          if (shouldVerifyCache && !isValidCacheContent(content)) {
```

Comment:
Ah- didn't realize we're caching both input (TS) and output (transpiled JS) files.